### ISSUE TITLE
'Delete this entry' on Single Types only ever deletes the default locale regardless of which locale is selected.

### ISSUE THREAD

**Issue by andycansdale** [strapi/strapi#10997]
## Bug report

'Delete this entry' on Single Types only ever deletes the default locale regardless of which locale is selected.

### Describe the bug

With localization enabled, deleting a translation for a single type always deletes the translation for the default locale even when viewing a different locale. Additionally, when subsequently trying to again delete a non-default locale, a 'Not Found' error message is shown, presumably because the translation for the default locale no longer exists.

### Steps to reproduce the behavior

1. Add a new single type and save.
2. Add translations for the same type and save.
3. Select any one of the translations for the non-default locale and select Delete this entry

The above is confirmed by changing the default locale and carrying out the same steps.

### Expected behavior

Only the translation for the selected locale should be deleted and not one for the default locale.

### System

STRAPI VERSION
v3.6.8

NODE VERSION
v14.16.0

**Comment by kasonde** [comment]
Hi @andycansdale, to work around this: it's not the cleanest of workarounds but this is what you can do to extend and modify the functionality of singleType deletion:

### TL;DR
1. Create a new file `./extenstions/content-manager/single-types.js`
2. Paste the following in:
```javascript
 "use strict";

const qs = require("qs");

const { getService } = require("strapi-plugin-content-manager/utils");

const findEntity = async (query, model) => {
  const entityManager = getService("entity-manager");
  const entity = await entityManager.find(query, model);
  return entityManager.assocCreatorRoles(entity);
};

module.exports = {
  async delete(ctx) {
    const { userAbility } = ctx.state;
    const { model } = ctx.params;
    const {
      request: {
        header: { referer },
      },
    } = ctx;
    const queryString = referer.split("?")[1];
    const {
      plugins: {
        i18n: { locale },
      },
    } = qs.parse(queryString);
    const { query = {} } = ctx.request;

    const entityManager = getService("entity-manager");
    const permissionChecker = getService("permission-checker").create({
      userAbility,
      model,
    });

    if (permissionChecker.cannot.delete()) {
      return ctx.forbidden();
    }

    const entity = await findEntity({ ...query, _locale: locale }, model);

    if (!entity) {
      return ctx.notFound();
    }

    if (permissionChecker.cannot.delete(entity)) {
      return ctx.forbidden();
    }

    const deletedEntity = await entityManager.delete(entity, model);

    ctx.body = permissionChecker.sanitizeOutput(deletedEntity);
  },
};
```
### Explanation
What I did was extend the content-manager controller file and basically copy-pasted the original code that performs the content deletion from the original plugin code. 

The query parameters are empty in the request so the line `const entity = await findEntity(query, model);` does not contain any information that targets the desired locale directly. To know what locale is required directly, I used the `referer` key on the request header. This contains the URL to the page that fired the request on the front end. 
```javascript
 const {
      request: {
        header: { referer },
      },
    } = ctx;
    const queryString = referer.split("?")[1];
    const {
      plugins: {
        i18n: { locale },
      },
    } = qs.parse(queryString);
```
This code takes the `queryString` (all text after the '?' in a URL) and parses it with `qs` to get an object I can traverse easily to get the locale parameter.

**Comment by thexpand** [comment]
I believe this happens not only on single types but on collection types as well. The day before yesterday I was bewildered why one of the UIDs was saying it was already in use (unavailable). It turned out there was an entry that was deleted but only from the default locale. By switching to the additional locale from the dropdown on the list page, I was able to find the entry that was not deleted.

**Comment by gu-stav** [comment]
Hi 👋🏼 I have looked into the issue using strapi@4.6.1 today and I was not able to reproduce it.

@thexpand on which version do you experience that problem?

**Comment by thexpand** [comment]
@gu-stav I'm experiencing this on strapi@4.6.1
Steps to reproduce this:
1. Create a new record by filling in the default locale.
2. Save the record in the default locale.
3. Switch to a different locale and start filling in the information.
4. Save the record in the new locale.
5. Go to the list and while viewing the default locale, mark the record and delete it.
6. Form this list, switch the locale to the other one, and see that the record from the other locale is not deleted.

In general, Strapi treats a record from each locale as a separate entity. By deleting the default locale, it doesn't delete the others.

**Comment by gu-stav** [comment]
@thexpand So this happens to you in the CM list-view for collection-types? I think I was confused about that because the original issue was about single-types which don't have a list.

**Comment by thexpand** [comment]
@gu-stav Sorry for the confusion. I am experiencing this with collection types.

**Comment by gu-stav** [comment]
@thexpand Gotcha - thanks. I was able to reproduce the issue with bulk actions.
